Enable feature "Episode ID Roles" by default

LukasKalbertodt opened the pull request:
This is a very useful feature to have, especially for external systems. Especially especially so with JWT, which depends on this to make the `oc` claim work. For external apps, it's extremely useful to be able to rely on the presence of this feature and it will be annoying to tell everyone to enable it manually. For example, Tobira is already making use of this.

CC @lkiesow @Arnei  (since you implemented this in #5056)


### How to test this patch

Build & start Opencast, then somehow create a user with a magic role (permanent or just a JWT session) and see that the user has access to the event.


### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [x] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

LukasKalbertodt committed:
Enable feature "Episode ID Roles" by default

This is a very useful feature to have, especially for external systems.
Especially especially so with JWT, which depends on this to make the
`oc` claim work. For external apps, it's extremely useful to be able
to rely on the presence of this feature and it will be annoying to tell
everyone to enable it manually. For example, Tobira is already making
use of this.

lkiesow commented:
Sometimes having less options (being more consistent) is nice. Would it hurt us to always enable this and remove the configuration completely?

Also, maybe we could mention in the release notes that you can do the reindexing at runtime while people are already using the system and the new roles will then be available eventually.

LukasKalbertodt commented:
> Sometimes having less options (being more consistent) is nice. Would it hurt us to always enable this and remove the configuration completely?

Hell yeah, I'm absolutely in favor of doing that. I was just not sure if that'd be fine for the community. If you think it's fine, I can adjust this PR.

> Also, maybe we could mention in the release notes that you can do the reindexing at runtime while people are already using the system and the new roles will then be available eventually.

I didn't know about that (in fact, I don't know any details about what exactly needs reindexing), but I can add it to the changelog.

LukasKalbertodt committed:
Remove config `episode.id.role.access` & always enable ID based access

This removes the configuration for this feature, making it always
enabled. Having less variability in these configurations makes it easier
for external apps to rely on Opencast behavior, which in turn makes
integration easier for admins. And this feature in particular will
likely be widely used by external apps via JWT auth.

LukasKalbertodt committed:
Mention information about ID-based auth index rebuild in changelog

LukasKalbertodt commented:
Rebased
